ArcGISRuntimeSDKQt_CppSamples/Routing/NavigateARouteWithRerouting/NavigateARouteWithRerouting.cpp

```diff
@@ -0,0 +1,369 @@
+    return;
+
+  m_mapView = mapView;
+  m_mapView->setMap(m_map);
+
+  m_mapView->graphicsOverlays()->append(m_routeOverlay);
+  connectRouteTaskSignals();
+
+  connect(m_routeTask, &RouteTask::loadStatusChanged, this, [this](LoadStatus loadStatus)
+  {
+    if (loadStatus == LoadStatus::Loaded)
+    {
+      // Request default parameters once the task is loaded
+      m_routeTask->createDefaultParameters();
+    }
+  });
+
+  connect(m_routeTask, &RouteTask::createDefaultParametersCompleted, this, [this](QUuid, RouteParameters routeParameters)
+  {
+    // Store the resulting route parameters
+    m_routeParameters = routeParameters;
+  });
+
+  m_routeTask->load();
+
+  // add graphics for the predefined stops
+  SimpleMarkerSymbol* stopSymbol = new SimpleMarkerSymbol(SimpleMarkerSymbolStyle::Diamond, Qt::red, 20, this);
+  m_routeOverlay->graphics()->append(new Graphic(conventionCenterPoint, stopSymbol, this));
+  m_routeOverlay->graphics()->append(new Graphic(aerospaceMuseumPoint, stopSymbol, this));
+
+  emit mapViewChanged();
+}
+
+void NavigateARouteWithRerouting::connectRouteTaskSignals()
+{
+  connect(m_routeTask, &RouteTask::solveRouteCompleted, this, [this](QUuid, RouteResult routeResult)
+  {
+    if (routeResult.isEmpty())
+      return;
+
+    if (routeResult.routes().empty())
+      return;
+
+    m_routeResult = routeResult;
+    m_route = qAsConst(m_routeResult).routes()[0];
+
+    m_directionManeuvers = m_route.directionManeuvers(this)->directionManeuvers();
+
+    // adjust viewpoint to enclose the route with a 100 DPI padding
+    m_mapView->setViewpointGeometry(m_route.routeGeometry(), 100);
+
+    // create graphics to show the route traversed and route ahead
+    m_routeAheadGraphic = new Graphic(m_route.routeGeometry(), new SimpleLineSymbol(SimpleLineSymbolStyle::Solid, Qt::blue, 5, this), this);
+    m_routeTraveledGraphic = new Graphic(Geometry(), new SimpleLineSymbol(SimpleLineSymbolStyle::Solid, Qt::cyan, 3, this), this);
+    m_routeOverlay->graphics()->append(m_routeAheadGraphic);
+    m_routeOverlay->graphics()->append(m_routeTraveledGraphic);
+
+    m_navigationEnabled = true;
+    emit navigationEnabledChanged();
+  });
+
+  connect(m_routeTask, &RouteTask::createDefaultParametersCompleted, this, [this](QUuid, RouteParameters defaultParameters)
+  {
+    // set values for parameters
+    defaultParameters.setReturnStops(true);
+    defaultParameters.setReturnDirections(true);
+    defaultParameters.setReturnRoutes(true);
+    defaultParameters.setOutputSpatialReference(SpatialReference::wgs84());
+
+    Stop stop1(conventionCenterPoint);
+    Stop stop2(aerospaceMuseumPoint);
+
+    QList<Stop> stopsList = {stop1, stop2};
+    defaultParameters.setStops(stopsList);
+
+    m_routeTask->solveRoute(defaultParameters);
+  });
+
+  connect(m_routeTask, &RouteTask::doneLoading, this, [this](const Error& error)
+  {
+    if (error.isEmpty())
+    {
+      m_routeTask->createDefaultParameters();
+    }
```

Comment:
This is duplicated with the above connection to `RouteTask::loadStatusChanged` where createDefaultParameters is ultimately called once loaded